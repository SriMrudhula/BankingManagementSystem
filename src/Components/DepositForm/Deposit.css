.err{
    font-size: 10px;
    color: red;
}
.img3{
    background-image: url('../../Asserts/banking.jpg');
    height: 503px;
    background-repeat: no-repeat;
    background-size: cover;
    margin-top: -90px;
    width:100%;
    background-attachment: fixed;
}
     .b2{
        display:inline-block;
        width: 400px;
        height: 250px;
        padding: 50px;
        background-color: rgba(0,0,0,0.5);
        background-size: 400px 480px;
        margin-left:40px;    
        margin-top:100px;
        position: relative;
         }
     label,.data{
       font-size: large;
        color: white;
        margin-left: 20px;
    }
    .btn{
        margin-left: auto;
        height:25px;
    width:75px;
    border-radius: 5px;
    }
    .btn1{
        margin-left: auto;
        height:50px;
    width:100px;
    border-radius: 10px;
    }
    input,select{
        margin-left: 30px;
         border-radius: 4px;
        width: 120px;
        height:25px; 

    }
    .heading{
        color: white;
        font-size: 30px;
    }