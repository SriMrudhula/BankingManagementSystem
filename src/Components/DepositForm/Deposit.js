import React, { Component } from 'react';
import './Deposit.css';
import axios from 'axios';
import { Redirect } from "react-router-dom";
import { Container } from '@material-ui/core';
import { Row, Col} from 'reactstrap';
class Deposit extends Component
{
  state={    
    users:[{
      id:"",
      name:"",
      username:"",
      password:"",
      address:"",
      country:"",
      state:"",
      email:"",
      contact:"",
      dob:"",
      accType:"",
      branchName:"",  
      initialDeposit:"",
      identificationProofType:""}
    ],
    deposit:"",
    redirect: null 
  }

  constructor(){
    super();
    if(localStorage.getItem("id")===null){
      this.state.redirect= "/logout";
    }
    this.state.users[0].id=localStorage.getItem("id");
    this.state.users[0].name= localStorage.getItem("name");
    this.state.users[0].username=localStorage.getItem("userName");
    this.state.users[0].password= localStorage.getItem("password");
    this.state.users[0].address=localStorage.getItem("address");
    this.state.users[0].country=localStorage.getItem("country");
    this.state.users[0].state=localStorage.getItem("state");
    this.state.users[0].email=localStorage.getItem("email");
    this.state.users[0].contact=localStorage.getItem("contact");
    this.state.users[0].dob=localStorage.getItem("dob");
    this.state.users[0].accType=localStorage.getItem("accType");
    this.state.users[0].branchName=localStorage.getItem("branchName");
    this.state.users[0].initialDeposit=localStorage.getItem("initialDeposit");
    this.state.users[0].identificationProofType=localStorage.getItem("identificationProofType");
    this.state.accountType=localStorage.getItem("accType"); 
  }

  handleChange =event=>{
    this.setState({
        [event.target.name]:event.target.value
    });
  }
       
  handleSubmit = event => {
    event.preventDefault();      
    this.state.users[0].initialDeposit=parseInt(this.state.users[0].initialDeposit,10)+parseInt(this.state.deposit,10);
    axios.put('http://localhost:3000/users/'+this.state.users[0].id,this.state.users[0])
    .then(respone =>{
      console.log(respone);
      alert("Your balance is " +this.state.users[0].initialDeposit);
      localStorage.setItem("initialDeposit",this.state.users[0].initialDeposit);
      this.setState({ redirect: "/home" });
    })
    .catch(error=>{
      console.log(error)
    })    
  }

  checkBalance=()=>{
    alert("Your Balance is "+this.state.users[0].initialDeposit);
    this.setState({ redirect: "/home" });
  }
  
  handleClick = event => {
    this.props.handleClick(event);     
  };

  render()
  {
    if (this.state.redirect) {
      return <Redirect to={this.state.redirect} />
    }
    return (  
      <div className="img3">
      <Row>
      <div className="b2">
      <div className="heading">Deposit</div>
      <Container>
      <form onSubmit={this.handleSubmit} >
      <div className="form-group">
      <Row>
      <Col>
      <label>Name</label>
      <input type="text" value={this.state.users[0].name} className="form-control" onChange={this.handleChange} name="name"/>
      </Col>
      <Col>
      <label>Account type</label>
      <select  value={this.state.users[0].accType} name="accType" className="form-control" onChange={this.handleChange} required>
      <option value="Salary Account">Salary Account</option>
      <option value="Saving Account">Saving Account</option>
      </select>
      </Col>
      </Row><br/>
      <Row>
      <label>Deposit Amount</label>
      <input type="number" value={this.state.deposit} name="deposit" className="form-control" onChange={this.handleChange} required/>
      </Row><br/>
      <button className="btn-primary">Deposit</button>
      </div>
      </form> 
      </Container>
      </div>
      <div className="b6">
      <form >
      <button className="btn1" type="submit" id="depositsubmit" onClick={this.checkBalance}>My Balance</button>
      </form> 
      </div>
      </Row>
      </div>
    )
    }
  }

export default Deposit;