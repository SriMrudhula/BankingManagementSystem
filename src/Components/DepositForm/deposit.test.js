import * as React from 'react';
import Enzyme,{shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import Deposit from './Deposit';
Enzyme.configure({ adapter: new Adapter() });
describe('Test when form is submitted', () => {
    it('should submit form data', () => {
        localStorage.setItem("id",10);
        let form = shallow(<Deposit/>); 
        form.instance().handleSubmit = jest.fn();
        form.instance().handleSubmit();
        form.update();                                                   
        form.find('#depositsubmit').simulate('submit');
        expect(form.find('#depositsubmit').length).toEqual(1);
        expect(form.instance().handleSubmit).toHaveBeenCalled();
    });    
    it('renders correctly', () => {
        const tree = renderer.create(<BrowserRouter><Deposit/></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
      });
});
