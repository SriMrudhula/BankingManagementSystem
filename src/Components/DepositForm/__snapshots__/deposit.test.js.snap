// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test when form is submitted renders correctly 1`] = `
<div
  className="img3"
>
  <div
    className="row"
  >
    <div
      className="b2"
    >
      <div
        className="heading"
      >
        Deposit
      </div>
      <div
        className="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                <label>
                  Name
                </label>
                <input
                  className="form-control"
                  name="name"
                  onChange={[Function]}
                  type="text"
                  value={null}
                />
              </div>
              <div
                className="col"
              >
                <label>
                  Account type
                </label>
                <input
                  className="form-control"
                  name="accType"
                  onChange={[Function]}
                  required={true}
                  type="text"
                  value={null}
                />
              </div>
            </div>
            <br />
            <div
              className="row"
            >
              <label>
                Deposit Amount
              </label>
              <input
                className="form-control"
                name="deposit"
                onChange={[Function]}
                required={true}
                type="number"
                value=""
              />
            </div>
            <br />
            <button
              className="btn-primary"
              id="depositsubmit"
            >
              Deposit
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      className="b6"
    >
      <form>
        <button
          className="btn1"
          onClick={[Function]}
          type="submit"
        >
          My Balance
        </button>
      </form>
    </div>
  </div>
</div>
`;
